subsys/bluetooth/controller/CMakeLists.txt

```diff
@@ -1,68 +1,73 @@
 zephyr_library()
-zephyr_library_sources(
-  util/mem.c
-  util/memq.c
-  util/mayfly.c
-  util/util.c
-  ticker/ticker.c
-  ll_sw/ll_addr.c
-  ll_sw/ll_tx_pwr.c
-  ll_sw/ctrl.c
-  ll_sw/ll.c
-  hci/hci_driver.c
-  hci/hci.c
-  )
 
-zephyr_library_sources_ifdef(
-  CONFIG_BT_CTLR_CRYPTO
-  crypto/crypto.c
-  )
+if(CONFIG_BT_LL_SW)
+  zephyr_library_sources(
+    util/mem.c
+    util/memq.c
+    util/mayfly.c
+    util/util.c
+    ticker/ticker.c
+    ll_sw/ll_addr.c
+    ll_sw/ll_tx_pwr.c
+    crypto/crypto.c
+    hci/hci_driver.c
+    hci/hci.c
+    )
+  zephyr_library_sources(
```

Comment:
That looks correct but this was how it was structured before so I didn't change any of this part. i.e., not sure why it was like this in the first place?

Reply:
You could make it similar to left hand side, as it is in the upstream master.

Reply:
I'm not following what you are requesting? I have to wrap these sources with the BT_LL_SW test because these files are not needed when building for the KW41 link layer library.

Reply:
in line 4 there is zephyr_library_sources, and you add it again in line 16. On the left hand side, in the parent commit, the ctrl.c and ll.c is part of the single zephyr_library_sources. 

also, you have removed CONFIG_BT_CTLR_CRYPTO dependency for the crypto.c file.